src/coordsim/trace_processor/trace_processor.py

```diff
@@ -11,23 +11,33 @@ class TraceProcessor():
 
     def __init__(self, params: SimulatorParams, env: Environment, trace: list,):
         self.params = params
+        self.params.inter_arr_means = []
         self.env = env
         self.trace_index = 0
         self.trace = trace
-        self.env.process(self.change_inter_arrival_time())
+        self.env.process(self.process_trace())
 
-    def change_inter_arrival_time(self):
+    def process_trace(self):
         """
         Changes the inter arrival mean during simulation
         The initial time is read from the the config file, so if the inter_arrival_time set in the trace CSV
         file does not start from 0, then the simulator will use the value set in sim_config
 
         """
         self.timeout = float(self.trace[self.trace_index]['time']) - self.env.now
-        inter_arrival_mean = float(self.trace[self.trace_index]['inter_arrival_mean'])
+        inter_arrival_mean = self.trace[self.trace_index]['inter_arrival_mean']
         yield self.env.timeout(self.timeout)
         log.debug(f"Inter arrival mean changed to {inter_arrival_mean} at {self.env.now}")
-        self.params.inter_arr_mean = inter_arrival_mean
+        if 'node' in self.trace[self.trace_index]:
+            node_id = self.trace[self.trace_index]['node']
+            if inter_arrival_mean == 'None':
+                self.params.inter_arr_mean[node_id] = None
```

Comment:
Did you test this with a sample trace file? Is `inter_arrival_mean` a dict or array? Probably won't work with an array, since node IDs/names are strings.

Could you also add an example trace that makes use of the new varying ingress feature?

Reply:
This is extra code that I forgot to delete :D Will delete it now and push a new commit

Reply:
I tested it and it was fine. I already added a trace file with varying ingress nodes in the other PR.